src/transformers/modeling_flax_utils.py

```diff
@@ -111,13 +112,33 @@ def __init__(
         # Those are public as their type is generic to every derived classes.
         self.key = PRNGKey(seed)
         self.dtype = dtype
+        self.input_shape = input_shape
 
-        # randomly initialized parameters
-        random_params = self.init_weights(self.key, input_shape)
+        # To check if the model was intialized automatically.
+        self._is_intialized = do_init
+
+        if do_init:
+            # randomly initialized parameters
+            random_params = self.init_weights(self.key, input_shape)
+            params_shape_tree = jax.eval_shape(lambda params: params, random_params)
```

Comment:
wouldn't it be faster to retrieve the shapes here from `random_params`? (if possible it'd be nice to not slow down anything if it doesn't help)

Reply:
But if it's not faster then good to leave it!